tests/Query/Filter/DoneDateField.test.ts

```diff
@@ -38,6 +38,22 @@ describe('done date', () => {
         testTaskFilterForTaskWithDoneDate(filter, null, true);
         testTaskFilterForTaskWithDoneDate(filter, '2022-04-15', false);
     });
+
+    it('in current week/month/year', () => {
+        currentPeriodsTestArray.forEach((p) => {
+            const filter = new DoneDateField().createFilterOrErrorMessage('done in current ' + p);
+            testTaskFilterForTaskWithDoneDate(filter, null, false);
+            testTaskFilterForTaskWithDoneDate(filter, moment().format('YYYY-MM-DD'), true);
+            testTaskFilterForTaskWithDoneDate(filter, moment().startOf(p).format('YYYY-MM-DD'), true);
+            testTaskFilterForTaskWithDoneDate(filter, moment().endOf(p).format('YYYY-MM-DD'), true);
+            testTaskFilterForTaskWithDoneDate(
+                filter,
+                moment().startOf(p).subtract(1, 'second').format('YYYY-MM-DD'),
+                false,
+            );
+            testTaskFilterForTaskWithDoneDate(filter, moment().endOf(p).add(1, 'second').format('YYYY-MM-DD'), false);
+        });
```

Comment:
Oh thanks! I thought about that but want aware about the 'set...'.

Can I set it to a precise date so that I also test week and month behavior?

The setting shall be put in a it() call?

Reply:
Ok I found the answer below ]]]]